src/components/timeline/Track.css

```diff
@@ -8,23 +8,67 @@
  */
 .timelineTrack {
   padding: 0;
-  border-top: 1px solid var(--grey-30);
   margin: 0;
-  box-shadow: 0 1px var(--grey-30);
 }
 
 .timelineTrackLocal {
   margin-left: 15px;
 }
 
 .timelineTrackRow {
+  /* This is the width of the blue left border when a track is selected.
+   * This variable will be used every time we need to calculate some width that
+   * depends on this value, from the intended perceived width. */
+  --selected-left-border-width: 3px;
+
   display: flex;
   flex-flow: row nowrap;
+
+  /* This padding is added to the button's padding as the perceived padding. For
+   * a global row, it will be replaced with a margin when it's selected. */
+  padding-left: var(--selected-left-border-width);
+  border-top: 1px solid var(--grey-30);
   background-color: #fff;
 }
 
 .timelineTrackRow.selected {
   background-color: #edf6ff;
+
+  /* We use a box-shadow on the track row instead of using border, so
+   * that when we select adjacent tracks, we see just one line, without any
+   * border on top of it. */
+  box-shadow: calc(-1 * var(--selected-left-border-width)) 0 var(--blue-60);
+}
+
+.timelineTrackGlobalRow.selected {
+  /* We replace the padding by some margin. Indeed the box-shadow used to draw
+   * the blue selected left border is drawn outside of the border box, so the
+   * 3px margin makes this space available. */
+  padding-left: 0;
+  margin-left: var(--selected-left-border-width);
+}
+
+/* In the active tab view, we don't want any margin or padding, because
+ * otherwise we lose the alignment with the other panels. Therefore we won't
+ * have the blue left border when a track is selected in this view. */
+.timelineTrackRow.activeTab {
+  padding: 0;
+  margin: 0;
+}
+
+.timelineTrackLocalRow {
+  border-left: 1px solid var(--grey-30);
```

Comment:
This was moved from `.timelineTrackLocalLabel`, for more consistency with the other borders.